### PR TITLE
[km200] Migrate from apache-commons-httpclient to jetty (#4435)

### PR DESCRIPTION
**Pull request by flo-02-mu**
The change migrates from old apache-commons-httpclient to jetty-httpclient for performing GET/POST calls on the Buderus km200 device. 
It fixes the issue of not being able to install the km200 binding anymore in recent openhab versions due to missing dependencies.

Additionally the base64 encoding/decoding is now done using java.util.base64 so that another dependency is being removed.

### CODE REVIEW

**Comment by cweitkamp** [comment]
Thank you very much for your first contribution. Do you mean jetty? Or jersey?

**Comment by flo-02-mu** [comment]
Sorry, jetty not jersey. :)

File: addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java [hunk 1]
```diff
@@ -46,126 +44,141 @@ public KM200Comm(KM200Device device) {
...
+            client.start();
+            for (int i = 0; i < maxNbrRepeats.intValue() && rCode != HttpStatus.NO_CONTENT_204; i++) {
+
+                // Create a method instance.
+                contentResponse = client.newRequest("http://" + remoteDevice.getIP4Address() + service)
+                        .method(HttpMethod.POST)
+                        .agent("TeleHeater/2.2.3")
+                        .accept("application/json")
+                        .content(new BytesContentProvider(data))
+                        .timeout(5000, TimeUnit.SECONDS)
```
[the remark this answers is not in the document]
Comment (flo-02-mu) [reply addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java]: Corrected to 5 seconds.

File: addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java [hunk 2]
```diff
@@ -46,126 +44,141 @@ public KM200Comm(KM200Device device) {
...
+
+                // Execute the method.
+                statusCode = contentresponse.getStatus();
+
                 // Release the connection.
-                if (method != null) {
-                    method.releaseConnection();
+                switch (statusCode) {
+                    case HttpStatus.OK_200:
+                        remoteDevice.setCharSet("UTF-8");
```
[the remark this answers is not in the document]
Comment (flo-02-mu) [reply addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java]: Done.

**Comment by flo-02-mu** [comment]
I gave it a try using the shared instance. In fact this made it nicer since the start() / stop() including exceptions could be removed. So far no issues occurred during runtime.

**Comment by cweitkamp** [comment]
Looks good so far. But I am afraid something went wrong with your code formatter. The sorting of the import switched the other way around. Can you check it?

**Pull request merged**